### ISSUE TITLE
Unable to set `ldb_method` to something else than `static`.

### ISSUE THREAD

**Issue by jzandbergen** [fortinetdev/terraform-provider-fortios#246]
Hi All,

I am using the v1.15.0 to set a `fortios_firewall_vip` and I seem to be unable to change the ldb_method to something else than `static`.

Given the following terraform:
```hcl
resource "fortios_firewall_vip" "vip" {                                         
  name = "test_virtual_server"                                                  
  type = "server-load-balance"                                                  
  dynamic_sort_subtable = true                                                  
  server_type = "tcp"                                                           
  extintf = "any"                                                               
  extip = "10.255.0.201"                                                        
  extport = "1337"                                                              
  ldb_method = "first-active"                                                         
                                                                                
  dynamic "realservers" {                                                       
    for_each = local.serverts                                                   
      content {                                                                 
      type="ip"                                                                 
      ip=realservers.value                                                      
      port="22"                                                                 
      status="active"                                                           
      healthcheck="enable"                                                      
      monitor = "\"test_health_check\""                                         
    }                                                                           
  }                                                                             
                                                                                
  monitor {                                                                     
    name="test_health_check"                                                    
  }                                                                             
}            
```

When I try to apply, the terraform run completes succesfully and says:
```
  # fortios_firewall_vip.vip will be updated in-place
  ~ resource "fortios_firewall_vip" "vip" {
        id                               = "test_virtual_server"
      ~ ldb_method                       = "static" -> "first-alive"
        name                             = "test_virtual_server"
        # (57 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }
```

But nothing changes in the fortigate itself.

Am I doing something wrong here?

Thanks!

**Comment by lix-fortinet** [comment]
Hi @jzandbergen,

Thank you for raising this issue. This issue will occur when setting argument `server_type` to `tcp` at the same time. Somehow, argument `ldb_method` could not be set to value other than `static` when argument `server_type`  also exist for the HTTP request. We have created an internal ticket to track this issue, and we will work with the related team to find a solution. We will get back to you once we have any update. 

Thanks,
Xing

**Comment by eddyfussel** [comment]
any updates on this?

**Comment by invizus** [comment]
Still does not work.
```
resource "fortios_firewall_vip" "example" {
...
  type        = "server-load-balance"
  server_type = "tcp"
  ldb_method  = "least-session"
...
}
```

**Comment by invizus** [comment]
Hello. I have just tested this with CURL, and it works. However I had to figure out correct JSON payload. The one I copied from Fortinet OS preview, I assumed payload should be:
```json
{
  "data": {
    "ldb-method": "round-robin",
    "extport": 443
  }
}
```
Running this command shows no change, in fact resets load balancing method to static.
```
url -k -XPUT -H "Authorization: Bearer xxxxxxx" --data @data.json https://1.2.3.4/api/v2/cmdb/firewall/vip/my-test-vip
{
  "http_method":"PUT",
  "revision":"xxxxxxxxxxxxxxxxxxxxxxxxxx",
  "revision_changed":false,
  "mkey":"my-test-vip",
  "status":"success",
  "http_status":200,
  "vdom":"root",
  "path":"firewall",
  "name":"vip",
  "serial":"FGXXXXXXXXXX",
  "version":"v7.2.7",
  "build":1577
}
```
However, when I changed payload to this, my curl command worked and I am able to change load balancer method.
Working JSON:
```json
{
    "ldb-method": "round-robin",
    "extport": 443
}
```

Could it be that terraform provider is sending payload in a wrong format?

**Comment by invizus** [comment]
@lix-fortinet maybe above comment is of any help?

**Comment by jzandbergen** [comment]
@lix-fortinet any updates on this issue?

**Comment by MaxxLiu22** [comment]
Hi @jzandbergen ,

Apologies for the delayed update. This issue is related to the FOS backend and has been addressed in FOS v7.4.4, build 2662, and v7.6.0, build 1401. For versions 7.0.x and 7.2.x, I have initiated an internal ticket with the API team. In the meantime, the ldb_method can be set alone successfully, providing a temporary workaround. I apologize for any inconvenience this may have caused.
```
variable "vip" {
  description = "firewall vip name"
  type        = string
  default     = "vip"
}

resource "fortios_firewall_vip" "trname" {
  extintf     = "any"
  extip       = "2.3.4.5"
  extport     = 23
  name        = var.vip
  server_type = "tcp"
  type        = "server-load-balance"
  ldb_method  = "least-session"
}


resource "fortios_json_generic_api" "test2" {
  path       = "/api/v2/cmdb/firewall/vip/${var.vip}"
  method     = "PUT"
  json       = <<EOF
{
  "ldb-method": "least-session"}
EOF
  depends_on = [fortios_firewall_vip.trname]
}
```

Thanks,
Maxx
